src/js/containers/LoginPage/LoginPage.js

```diff
@@ -17,19 +37,126 @@ class LoginPage extends Component {
         }
     }
 
+    openLinkInWindow(link) {
+        window.open(link, '_blank', 'toolbar=0,status=0,width=720,height=700');
+    }
+
+    handleEmailChange(email) {
+        this.setState({ email });
+    }
+
+    handleApiKeyChange(apiKey) {
+        this.setState({ apiKey });
+    }
+
+    handleLoginSubmit(e) {
+        e.preventDefault();
+        const { dispatch, config } = this.props;
+
+        let isHostAPILogin = getConfigValue(config, 'useHostAPILogin');
+        if(isHostAPILogin) {
+            //apiKey will actually be password here
+            dispatch(asyncLogin(this.state.email , this.state.apiKey));
+        } else {
+            dispatch(asyncAPILogin(this.state.email, this.state.apiKey));
+        }
+    }
+
     render() {
+        const { formatMessage } = this.props.intl;
         const { config } = this.props;
-        let isHostAPILogin = getConfigValue(config, 'useHostAPILogin');
+
+        //client login defaults
+        let title = 'component.clientLogin.form.title';
+        let inputType = 'text';
+        let inputLabel = 'component.clientLogin.form.apiKey';
+        let loginButtonText = 'component.clientLogin.form.button';
+
+        const isHostAPILogin = getConfigValue(config, 'useHostAPILogin');
+
+        if(isHostAPILogin) {
+          title = 'component.login.form.title';
+          inputType = 'password';
```

Comment:
I'll refactor apiKey to read secret, I combined the two login pages into one so if its the host API login they enter their password but we set it to the apiKey field.